plugins/@grouparoo/sendgrid/__tests__/export/export-profile.ts

```diff
@@ -0,0 +1,483 @@
+      if (user) {
+        usersIdsToDelete.push(user["id"]);
+      }
+    }
+    if (usersIdsToDelete.length > 0) {
+      await apiClient.deleteUsers(usersIdsToDelete);
+    }
+  } catch (err) {
+    if (!suppressErrors) {
+      throw err;
+    }
+  }
+}
+
+async function deleteLists(suppressErrors) {
+  try {
+    for (const groupToDelete of [listOne, listTwo, listThree, listFour]) {
+      const listId = await getListId(groupToDelete);
+      if (listId) {
+        await apiClient.deleteList(listId);
+      }
+    }
+  } catch (err) {
+    if (!suppressErrors) {
+      throw err;
+    }
+  }
+}
+
+async function cleanUp(suppressErrors) {
+  await deleteUsers(suppressErrors);
+  await deleteLists(suppressErrors);
+  await indexContacts(newNock, 30 * 1000);
+}
+
+async function runExport({
+  oldProfileProperties,
+  newProfileProperties,
+  oldGroups,
+  newGroups,
+  toDelete,
+}) {
+  return exportProfile({
+    appOptions,
+    appId,
+    connection: null,
+    app: null,
+    destination: null,
+    destinationId: null,
+    destinationOptions: null,
+    export: {
+      profile: null,
+      profileId: null,
+      oldProfileProperties,
+      newProfileProperties,
+      oldGroups,
+      newGroups,
+      toDelete,
+    },
+  });
+}
+
+describe("sendgrid/exportProfile", () => {
+  beforeAll(async () => {
+    apiClient = await connect(appOptions);
+    await cleanUp(false);
+  }, helper.setupTime);
+
+  afterAll(async () => {
+    await cleanUp(true);
+  }, helper.setupTime);
+
+  test("can create profile on Sendgrid", async () => {
+    user = await apiClient.getUser(email);
+    expect(user).toBe(null);
+
+    await runExport({
+      oldProfileProperties: {},
+      newProfileProperties: { email, first_name },
+      oldGroups: [],
+      newGroups: [],
+      toDelete: false,
+    });
+    await indexContacts(newNock, 60 * 1000);
+
+    user = await apiClient.getUser(email);
+
+    expect(user).not.toBe(null);
+    expect(user["email"]).toBe(email);
+    expect(user["first_name"]).toBe(first_name);
+  });
+
+  test("can add user variables", async () => {
+    await runExport({
+      oldProfileProperties: { email, first_name },
+      newProfileProperties: {
+        email,
+        first_name,
+        last_name,
+        city,
+        phone_number,
+      },
+      oldGroups: [],
+      newGroups: [],
+      toDelete: false,
+    });
+    await indexContacts(newNock, 60 * 1000);
+
+    const user = await apiClient.getUser(email);
+    expect(user.first_name).toBe(first_name);
+    expect(user.last_name).toBe(last_name);
+    expect(user.city).toBe(city);
+    expect(user.phone_number).toBe(phone_number);
+  });
+
+  test("can change user variables", async () => {
+    // Phone must be valid.
+    await runExport({
+      oldProfileProperties: {
+        email,
+        first_name,
+        phone_number,
+      },
+      newProfileProperties: {
+        email,
+        first_name: alternativeName,
```

Comment:
we want to make sure to try one of each type here.
Date, Number, etc.
And then verify it worked.